src/MetadataUtility/Metadata/SupportFiles/FrontierLabs/LogFile.cs

```diff
@@ -246,7 +251,7 @@ public static Microphone MicrophoneParser(string value)
         /// <returns>
         /// A string containing only numeric characters.
         /// </returns>
-        public static string NumericParser(string value) => new string(value.Where(c => char.IsDigit(c) || (new char[] { '.', '-', '+' }).Contains(c)).ToArray());
+        public static string NumericParser(string value) => Regex.Replace(value, "[^0-9+-. ]", string.Empty);
```

Comment:
OK this doesn't actually improve on the feedback. It is way more efficient though so I'll give you that.

A numeric parser that strips characters can do funny things to numbers. It almost always means that you're being imprecise with what you're trying to convert which can lead to ambiguities. Here are some examples:

```csharp
NumericParser(" 94% ( 4.13 V )") ➡️ "944.13"
NumericParser(" Ch A: 006277 \"STD AUDIO MIC\" ( 16/01/2020 )") ➡️ "00627716012020"
// scenario: log format changes from
//    "Temperature: 37 C"
// to 
//    "Humidity 50%. Temperature: 37 C"
NumericParser("Humidity 50%. Temperature: 37 C") ➡️ "50.37"

```